test/test_fx.py

```diff
@@ -2307,6 +2309,98 @@ def f():
         fx_f = symbolic_trace(f, enable_cpatching=True)
         assert(any(i.target == torch.randn for i in fx_f.graph.nodes))
 
+    def test_pytree(self):
+        def f_sum(x):
+            return sum(x)
+
+        def f_sum_dict(x):
+            out = 0
+            for k, v in x.items():
+                out += v
+            return out
+
+        def f_dict_list_map(x):
+            new_dict = {}
+            for k, v in x.items():
+                new_dict[k] = [i + 1 for i in v]
+            return new_dict
+
+        def f_dict_add(x):
+            return x['a'] + sum(x['z'])
+
+        class Foo(object):  # noqa: B209
+            def __init__(self, a, b):
+                self.a = a
+                self.b = b
+
+        pytree._register_pytree_node(
```

Comment:
I think so, yes.

For what it's worth, the ideas for how to handle `concrete_args` and the assert statements also run into this problem for "non-standard" objects. There, one way of handling it might be to

One way of resolving this is to never include the `self._in_spec` stuff on the graphmodule, and to always just bake it into the resulting FX code.

So `nf = fx.symbolic_trace(nf, concrete_args={'x': inp})` would also print out.
```
def forward(self, x):
    getitem = x['a']
    getitem_1 = x['b'];  x = None
    getitem_2 = getitem_1[0]
    getitem_3 = getitem_1[1];  getitem_1 = None
    add = 0 + getitem_2;  getitem_2 = None
    add_1 = add + getitem_3;  add = getitem_3 = None
    add_2 = getitem + add_1;  getitem = add_1 = None
    return add_2
```

Another is to just explicitly not support these kind of pytrees while serializing.

Reply:
Resolved from offline discussion - we will leave it up the user to make sure the pytrees are registered. I did add a relevant error message, though.